design/client_conformance_profiles.md

```diff
@@ -0,0 +1,550 @@
+# Client Conformance Profiles
+
+* **Status**: Draft #1
+* **JIRA**: [KEYCLOAK-11612](https://issues.jboss.org/browse/KEYCLOAK-11612)
+
+## Motivation
+
+To protect APIs that require high level security like providing financial services, there are several API security profiles like Financial-grade API Security Profile.
+
+To enforce the client operating in complying with such the security profile, the client conformance profiles are introduced. 
+
+## Overview

```

Comment:
I would also add more information about how to enforce profiles for resource servers. I mean, when the client in Keycloak is actually serving protected resources.

As an example, FAPI Read and Write defines some holder-of-keys mechanisms to prove that a token is actually bound to a caller when accessing a resource server. If we have plans to support that OOTB in our adapters, we would need to somehow configure the adapter properly to also conform with the specs.

I can understand if you think this is out of scope and we should only worry about the policy enforcement on Keycloak side. But something to consider when delivering a E2E deployment that is compliant with a profile.

Reply:
I've recognized that the client and the resource server also need to operate in comply with FAPI Security Profile. FAPI Security Profile specification describes requirements against the client and resource server.

At this time, I planned to only treat the authorization server, namely keycloak. Therefore, my proposal is as follows :

* [Scope] It is explicitly stated that this design document only treats the authorization server, namely keycloak.

* [FAPI Security Profile Support for keycloak client adapter] If there is a demand for FAPI Security Profile Support for keycloak client adapter and need to be supported, it might be better that the separate design document for FAPI Security Profile Support for keycloak client adapter  be newly created.

WDYT?

Reply:
Looks good. We should make clear the scope so that people take care of their deployments accordingly in order to adhere to the server policies / requirements.

Reply:
OK. I'll revise the design document to state these scopes explicitly. Thanks.